BaseApp/COD50012.TXT

```diff
@@ -0,0 +1,71 @@
+OBJECT Codeunit 50012 DotNet_HashAlgorithm
+{
+  OBJECT-PROPERTIES
+  {
+    Date=11/13/18;
+    Time=12:07:12 PM;
+    Modified=Yes;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    OnRun=BEGIN
+          END;
+
+  }
+  CODE
+  {
+    VAR
+      DotNetHashAlgorithm@1000000000 : DotNet "'mscorlib'.System.Security.Cryptography.HashAlgorithm";
+
+    [External]
+    PROCEDURE Create@1000000005(HashAlgorithm@1000000000 : 'SHA1,MD5,SHA256,SHA384,SHA512');
+    BEGIN
+      IF (HashAlgorithm < 0) OR (HashAlgorithm > 4) THEN
+        EXIT;
+      DotNetHashAlgorithm := DotNetHashAlgorithm.Create(FORMAT(HashAlgorithm));
+    END;
+
+    [External]
+    PROCEDURE ComputeHashFromText@1000000000(VAR Text@1000000001 : Text;VAR DotNet_Array@1000 : Codeunit 3000);
```

Comment:
If the intention is to wrap [ComputeHash(Byte[])](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm.computehash?view=netframework-4.7.2#System_Security_Cryptography_HashAlgorithm_ComputeHash_System_Byte___), why don't you pass an array of bytes?

Reply:
there are situations where the data in the calling code will reside in a stream (uploaded file, etc) so this interface makes the calling code simpler (no need to allocate and fill an array).  however, your point is valid, there are also situations where the data may reside in an array -- so I have added the "ComputeHash(DotNet_Array, ...)" interface per your comment -- please confirm this is what you expected